src/components/common/state-machine/state-machine.ts

```diff
@@ -754,6 +776,15 @@ const authStateMachine = createMachine(
         context.mfaMethodType === MFA_METHOD_TYPE.SMS &&
         context.support2FABeforePasswordReset === true,
       isPasswordChangeRequired: (context) => context.isPasswordChangeRequired,
+      is2FASMSPasswordChangeRequired: (context) =>
+        context.isPasswordChangeRequired === true &&
+        context.mfaMethodType === MFA_METHOD_TYPE.SMS &&
+        context.support2FABeforePasswordReset === true &&
+        context.requiresTwoFactorAuth === true,
```

Comment:
sms has context.requiresTwoFactorAuth === true but auth_app doesn't, is that correct?

Reply:
Nice spot, thanks. Must've removed it by accident.

Reply:
Done